aspnetcore/signalr/authn-and-authz.md

```diff
@@ -63,15 +63,7 @@ If [Windows authentication](xref:security/authentication/windowsauth) is configu
 
 Add a new class that implements `IUserIdProvider` and retrieve one of the claims from the user to use as the identifier. For example, to use the "Name" claim (which is the Windows username in the form `[Domain]\[Username]`), create the following class:
 
-```csharp
-public class NameUserIdProvider : IUserIdProvider
-{
-    public string GetUserId(HubConnectionContext connection)
-    {
-        return connection.User?.FindFirst(ClaimTypes.Name)?.Value;
-    }
-}
-```
+[!code-csharp[Identity.Name-based provider](authn-and-authz/sample/nameuseridprovider.cs?range=5-11)]
```

Comment:
Prefer "?name=region-name-here" over "?range=". This requires you to wrap the code in a region in the sample code. It makes things more maintainable moving forward. Line ranges are often not updated as the sample evolves.